src/server/game/Handlers/TaxiHandler.cpp

```diff
@@ -250,15 +251,11 @@ void WorldSession::HandleMoveSplineDoneOpcode(WorldPacket& recvData)
     // at this point only 1 node is expected (final destination)
     if (GetPlayer()->m_taxi.GetPath().size() != 1)
```

Comment:
this OpCode is triggered by vehicles too, which means this will always return true for vehicles...